kafka-connect-s3/src/main/java/io/confluent/connect/s3/TopicPartitionWriter.java

```diff
@@ -127,7 +128,13 @@ public TopicPartitionWriter(TopicPartition tp,
     rotateScheduleIntervalMs =
         connectorConfig.getLong(S3SinkConnectorConfig.ROTATE_SCHEDULE_INTERVAL_MS_CONFIG);
     if (rotateScheduleIntervalMs > 0) {
-      timeZone = DateTimeZone.forID(connectorConfig.getString(PartitionerConfig.TIMEZONE_CONFIG));
+      String timeZoneString = connectorConfig.getString(PartitionerConfig.TIMEZONE_CONFIG);
+      if (StringUtils.isBlank(timeZoneString)) {
+        throw new ConfigException(PartitionerConfig.TIMEZONE_CONFIG,
+                timeZoneString, "Timezone cannot be empty when using scheduled file rotation."
+        );
+      }
+      timeZone = DateTimeZone.forID(timeZoneString);
```

Comment:
@cricket007 I would make this check in S3SinkTask#start() [here](https://github.com/confluentinc/kafka-connect-storage-cloud/blob/master/kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkTask.java#L92). We want to check configuration before the task initiates any real work.